Code/Tools/AWSNativeSDKInit/source/AWSNativeSDKInit.cpp

```diff
@@ -79,6 +82,17 @@ namespace AWSNativeSDKInit
         Platform::CustomizeSDKOptions(m_awsSDKOptions);
         Aws::InitAPI(m_awsSDKOptions);
 
+        // Default to preventing calls to the Amazon EC2 instance metadata service (IMDS), unless environment var has been configured.
+        // AWS C++ SDK may reach out to EC2 IMDS for region, config or credentials, but unless code is running on EC2 compute
+        // such calls will fail and waste network resources.
+        // Note: AWS C++ SDK explicitly only checks if lower case version of AWS_EC2_METADATA_DISABLED == "true", otherwise it will enable the
+        // EC2 metadata service calls.
+        const auto ec2MetadataDisabled = Aws::Environment::GetEnv(AWS_EC2_METADATA_DISABLED);
```

Comment:
do we want to create a corresponding CVAR for users to set in order toggle this back on? Or would the guidance be to unset the env var directly in the underlying environment?

Reply:
discussed offline: CVar would be somewhat duplicative and managing both would be cumbersome. Guidance will be to unset the env var directly in the underlying environment.